### PR TITLE
Wire into signon entrypoints

### LINKED ISSUE #107: Firefox entry points to open Lockbox
Now that we have [Bug #1529729](https://bugzilla.mozilla.org/show_bug.cgi?id=1529729), we need to ensure that all the entry points (both new and old) appropriately open the Lockbox management interface.

## Implementation

* Add a method on the `experiments.logins` API to register a URL
* Call the `experiments.logins` API to make the registration
* Update the `signon.management.overrideURL` as a default

### PR DESCRIPTION
**Pull request by linuxwolf**
Fixes #107

Wires the Lockbox management UI into the built-in password manager entrypoints.  Anywhere Firefox presents the "saved logins" (or "logins and passwords") action, invoking it takes the user to Lockbox' management UI as a tab instead of the built-in password management UI.

## Testing and Review Notes

_(**Required**: steps to take to confirm this works as expected or other guidance for code, UX, and any other reviewers)_

## Screenshots or Videos

Once installed, the addon wires in the `signon.management.overrideURI` preference to the management page.

**From the main menu**:   addon-107 mainmenu-entrypoint

**From the 'saved logins' autofill selector**:  addon-107 autofill-entrypoint

**From the 'saved logins' context menu**:  addon-107 contextmenu-entrypoint

**From autofill -- no matching entries**:  addon-107 entrypoint-nomatches

## To Do

- [x] double check the original issue to confirm it is fully satisfied
- [x] add testing notes and screenshots in PR description to help guide reviewers
- [x] add unit tests and functional tests
- [x] make sure CI builds are passing (e.g.: fix lint and other errors)

### CODE REVIEW

File: src/experiments/logins/api.js [hunk 1]
```diff
@@ -63,6 +65,19 @@ this.logins = class extends ExtensionAPI {
               throw new ExtensionError(ex);
             }
           },
+          setManagementURI(url) {
+            try {
+              const prefs = Services.prefs.getDefaultBranch(null);
+              prefs.setStringPref(PREF_MANAGEMENT_OVERRIDE, url);
```
Comment (6a68) [inline src/experiments/logins/api.js]: I'd probably just do `Services.prefs.setStringPref()` directly, and not bother getting the branch on the line above.

File: src/background/environment.js [hunk 2]
```diff
@@ -11,4 +11,7 @@ export async function initializeEnvironment() {
     await browser.experiments.logins
       .setLoginSavingEnabled(origin, false);
   }
+
+  const mgmtURI = browser.extension.getURL("/list/manage.html?filter=%DOMAIN%");
```
Comment (6a68) [inline src/background/environment.js]: Might be good to document the fact that the internal desktop code replaces `%DOMAIN%` when `LoginHelper.openPasswordManager()` is called

File: src/experiments/logins/api.js (hunk 1 above)
Comment (linuxwolf) [reply src/experiments/logins/api.js]: The `getDefaultBranch()` call was recommended by MattN to help minimize the amount of cleanup the extension would need to do.

File: src/background/environment.js (hunk 2 above)
Comment (linuxwolf) [reply src/background/environment.js]: very good call; especially since that's not hooked up yet (follow on issue, or attempt to apply here?)

**Comment by linuxwolf** [comment]
@mozilla-lockbox/ux @mozilla-lockbox/product 
Screen captures in the description demonstrate the current state of this PR.  Note the results (right now) if there are no matching entries.  The filtering support can be reverted for now if that is too far from the desired state.

**Comment by changecourse** [comment]
@linuxwolf is that best demonstrated in the final screen capture? I think that's fine by me for now.

**Comment by linuxwolf** [comment]
> @linuxwolf is that best demonstrated in the final screen capture? I think that's fine by me for now.

Yes, the "no matching entries from 'autofilter'" (for lack of a better way to say it), is the one I'm most concerned about.  That is from our current codebase, before touching on #31 or #36 much.

It would not be too difficult to add/change the right-hand details to be some sort of instructional, if desired, or leave it blank (+ fxa/device promotion banner).

**Comment by linuxwolf** [comment]
would appreciate another look through, especially the added functional tests.

File: src/experiments/logins/api.js (hunk 1 above)
Comment (6a68) [reply src/experiments/logins/api.js]: TIL :-)

File: src/background/environment.js (hunk 2 above)
Comment (6a68) [reply src/background/environment.js]: not sure, looks like it's hooked up as of 67, at least https://bugzilla.mozilla.org/show_bug.cgi?id=1529729

Comment (linuxwolf) [reply src/background/environment.js]: by "not hooked up yet", I meant here in the extension ... at the time I wrote that comment (-:

**Pull request merged**